flixel/input/gamepad/FlxGamepadManager.hx

```diff
@@ -421,12 +423,35 @@ class FlxGamepadManager implements IFlxInputManager
 			
 			if (id >= 0)
 			{
-				var gamepad:FlxGamepad = createByID(id);
+				var gamepad:FlxGamepad = createByID(id, getModelFromFlashDeviceName(Device.name));
 				gamepad._device = Device;
 			}
 		}
 	}
 	
+	private function getModelFromFlashDeviceName(str:String):GamepadModel
+	{
+		str = str.toLowerCase();
+		var strip = ["-", "_"];
+		for (s in strip)
+		{
+			while (str.indexOf(s) != -1)
+			{
+				str = StringTools.replace(str, s, "");
+			}
+		}
+		
+		if (str.indexOf("xbox") != -1) return Xbox;					//"Microsoft X-Box 360 pad"
```

Comment:
This is why it's recommended to use spaces instead of tabs for formatting - as soon as the tab width changes, like here on GitHub, it looks horrible. ;)